basicdataset/src/main/java/ai/djl/basicdataset/tabular/TabularDataset.java

```diff
@@ -44,6 +44,13 @@ public TabularDataset(BaseBuilder<?> builder) {
         super(builder);
         features = builder.features;
         labels = builder.labels;
+
```

Comment:
Avoid throw exception in Constructor. Move this logic into `validate()`

Reply:
We had a `validate()` function, but the difficulty is that there wasn't a mechanism to ensure that the datasets inhering `TabularDataset` would call validate(). Resulting from this, even we did not remember to validate in all of them. Besides that, for most users they will get the exception as a result of calling `datasetBuilder.build()` regardless. Is there some other way you suggest implementing this?